File: mynotes-spring/src/main/kotlin/com/example/mynotes/security/auth/AuthenticationResponse.kt
package com.example.mynotes.security.auth

import com.example.mynotes.security.user.Role
import org.springframework.http.HttpEntity
data class AuthenticationResponse(
    val token: String,
    val userId: Long,
    val username: String,
    val email: String,
    val role: Role
) : HttpEntity<Map<String, Any>>(mapOf("token" to token, "userId" to userId, "username" to username, "email" to email, "role" to role))

File: mynotes-flutter/lib/services/auth/authentication_response.dart
import 'package:flutter/cupertino.dart';

@immutable
class AuthenticationResponse {
  final String token;
  final int userId;
  final String username;
  final String email;
  final String role;
  const AuthenticationResponse(
      {required this.token,
      required this.userId,
      required this.username,
      required this.email,
      required this.role});

  factory AuthenticationResponse.fromJson(Map<String, dynamic> json) {
    return AuthenticationResponse(
      token: json['token'] as String,
      userId: json['userId'] as int,
      username: json['username'] as String,
      email: json['email'] as String,
      role: json['role'] as String,
    );
  }
}
